browser/js/common/factories/single_item_factories.js

```diff
@@ -2,22 +2,29 @@
 app.factory("SingleItemFactory", function($http){
```

Comment:
Should be in ItemFactory, not SingleItemFactory. Can you combine the two into ItemFactory?